gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -495,6 +499,139 @@ impl GatewayLightning for ClnRpcService {
+    }
+
+    async fn open_channel(
+        &self,
+        request: tonic::Request<OpenChannelRequest>,
+    ) -> Result<tonic::Response<EmptyResponse>, Status> {
+        let request_inner = request.into_inner();
+
+        self.rpc_client()
+            .await
+            .map_err(|e| Status::internal(e.to_string()))?
+            .call(cln_rpc::Request::FundChannel(
+                model::requests::FundchannelRequest {
+                    id: cln_rpc::primitives::PublicKey::from_str(&request_inner.pubkey).map_err(
+                        |e| {
+                            error!("cln fundchannel pubkey parse error {:?}", e);
+                            tonic::Status::invalid_argument(e.to_string())
+                        },
+                    )?,
+                    amount: cln_rpc::primitives::AmountOrAll::Amount(
+                        cln_rpc::primitives::Amount::from_sat(request_inner.channel_size_sats),
+                    ),
+                    feerate: None,
+                    announce: None,
+                    minconf: None,
+                    push_msat: Some(cln_rpc::primitives::Amount::from_sat(
+                        request_inner.push_amount_sats,
+                    )),
+                    close_to: None,
+                    request_amt: None,
+                    compact_lease: None,
+                    utxos: None,
+                    mindepth: None,
+                    reserve: None,
+                },
+            ))
+            .await
+            .map_err(|e| {
+                error!("cln fundchannel rpc returned error {:?}", e);
+                tonic::Status::internal(e.to_string())
+            })?;
+
+        Ok(tonic::Response::new(EmptyResponse {}))
+    }
+
+    async fn wait_for_chain_sync(
+        &self,
+        request: tonic::Request<WaitForChainSyncRequest>,
+    ) -> Result<tonic::Response<EmptyResponse>, Status> {
+        let target_block_height = request.into_inner().block_height;
+
+        for _ in 0..MAX_INFO_RETRIES {
+            let blockheight = self
+                .rpc_client()
+                .await
+                .map_err(|e| Status::internal(e.to_string()))?
+                .call(cln_rpc::Request::Getinfo(
+                    model::requests::GetinfoRequest {},
+                ))
+                .await
+                .map(|response| match response {
+                    cln_rpc::Response::Getinfo(model::responses::GetinfoResponse {
+                        blockheight,
+                        ..
+                    }) => Ok(blockheight),
+                    _ => Err(ClnExtensionError::RpcWrongResponse),
+                })
+                .map_err(|e| {
+                    error!("cln getinfo rpc returned error {:?}", e);
+                    tonic::Status::internal(e.to_string())
+                })?
+                .map_err(|e| tonic::Status::internal(e.to_string()))?;
+
+            if blockheight >= target_block_height {
```

Comment:
Do you need to check if the chain is synced here like you did for LND?

Reply:
I didn't see a way to do this in CLN but after looking closer, it appears you're right! We're now also waiting for `warning_bitcoind_sync` to be `None`.